test/models/drive_test.dart

```diff
@@ -0,0 +1,489 @@
+        seats: 1,
+      );
+      expect(drive.isRidePossible(ride), false);
+    });
+    test('share ride with other rides', () async {
+      final Drive drive = DriveFactory().generateFake(
+        startTime: DateTime(2022, 2, 2, 14),
+        endTime: DateTime(2022, 2, 2, 16),
+        seats: 4,
+        rides: <Ride>[
+          RideFactory().generateFake(
+              startTime: DateTime(2022, 2, 2, 14, 20),
+              endTime: DateTime(2022, 2, 2, 15),
+              status: RideStatus.approved,
+              seats: 2),
+          RideFactory().generateFake(
+              startTime: DateTime(2022, 2, 2, 14, 40),
+              endTime: DateTime(2022, 2, 2, 14, 50),
+              status: RideStatus.approved,
+              seats: 1),
+          RideFactory().generateFake(
+              startTime: DateTime(2022, 2, 2, 15),
+              endTime: DateTime(2022, 2, 2, 15, 30),
+              status: RideStatus.approved,
+              seats: 2),
+        ],
+      );
+      final Ride ride = RideFactory().generateFake(
+          startTime: DateTime(2022, 2, 2, 14, 50),
+          endTime: DateTime(2022, 2, 2, 15, 10),
+          status: RideStatus.approved,
+          seats: 1);
+      expect(drive.isRidePossible(ride), true);
+    });
+  });
+  group('Drive.cancel', () {
+    test('cancelled is being changed from false to true', () async {
+      final Drive drive = DriveFactory().generateFake(cancelled: false, createDependencies: false);
+      when.call(driveProcessor.processUrl(any)).thenReturn(drive.toString());
+      drive.cancel();
+      expect(drive.cancelled, true);
+    });
+  });
+  group('Drive.toString', () {
+    test('returns a string representation of the drive', () async {
+      final Drive drive = DriveFactory().generateFake(
+        id: 1,
+        start: 'start',
+        startTime: DateTime.parse('2022-02-02T00:00:00.000Z'),
+        end: 'end',
+        endTime: DateTime.parse('2023-03-03T00:00:00.000Z'),
+        driverId: 5,
+        createDependencies: false,
+      );
+      expect(
+        drive.toString(),
+        'Drive{id: 1, from: start at 2022-02-02 00:00:00.000Z, to: end at 2023-03-03 00:00:00.000Z, by: 5}',
+      );
+    });
+  });
+  group('Drive.equals', () {
+    final Drive drive0 = DriveFactory().generateFake(
+      id: 1,
+      createdAt: DateTime(2022, 10),
+      start: 'Berlin',
+      startPosition: Position(1, 1),
+      startTime: DateTime(2022, 11),
+      end: 'Frankfurt',
+      endPosition: Position(2, 2),
+      endTime: DateTime(2022, 12),
+      seats: 1,
+      cancelled: false,
+      driverId: 2,
+      createDependencies: false,
+    );
+    test('same drive', () async {
+      expect(drive0.equals(drive0), true);
+    });
+    test('handle ride', () async {
+      final Ride ride = RideFactory().generateFake(createDependencies: false);
+      expect(drive0.equals(ride), false);
+    });
+    test('different cancelled', () async {
```

Comment:
I don't like this test. Either we test this through for every parameter (which I wouldn't do), or we just do it for one, and I think that's preferably the ID. The case can be made that it is important to show that even drives with the same ID are different to us if they have different values, but then why cancelled? I think start or something less obscure would make more sense in that case (you already have driverId below, that's fine too)

Reply:
that is covered by the equals test in trip. since euquals in drive and ride call their super from trip. I only check for the extra checks in drive and ride and the core trip values in Trip.equals